internal/controller/ols_app_server_assets.go

```diff
@@ -244,6 +251,96 @@ func (r *OLSConfigReconciler) generateService(cr *olsv1alpha1.OLSConfig) (*corev
 	return &service, nil
 }
 
+func (r *OLSConfigReconciler) generateServiceMonitor(cr *olsv1alpha1.OLSConfig) (*monv1.ServiceMonitor, error) {
+	metaLabels := generateAppServerSelectorLabels()
+	metaLabels["monitoring.openshift.io/collection-profile"] = "full"
+	metaLabels["app.kubernetes.io/component"] = "metrics"
+
+	serviceMonitor := monv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      AppServerServiceMonitorName,
+			Namespace: r.Options.Namespace,
+			Labels:    metaLabels,
+		},
+		Spec: monv1.ServiceMonitorSpec{
+			Endpoints: []monv1.Endpoint{
+				{
+					Port:     "https",
+					Path:     AppServerMetricsPath,
+					Interval: "30s",
+					Scheme:   "https",
+					TLSConfig: &monv1.TLSConfig{
+						CAFile:   "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt",
+						CertFile: "/etc/prometheus/secrets/metrics-client-certs/tls.crt",
+						KeyFile:  "/etc/prometheus/secrets/metrics-client-certs/tls.key",
+						SafeTLSConfig: monv1.SafeTLSConfig{
+							InsecureSkipVerify: false,
+							ServerName:         strings.Join([]string{OLSAppServerServiceName, r.Options.Namespace, "svc"}, "."),
+						},
+					},
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+				},
+			},
+			JobLabel: "app.kubernetes.io/name",
+			Selector: metav1.LabelSelector{
+				MatchLabels: generateAppServerSelectorLabels(),
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(cr, &serviceMonitor, r.Scheme); err != nil {
+		return nil, err
+	}
+
+	return &serviceMonitor, nil
+}
+
+func (r *OLSConfigReconciler) generatePrometheusRule(cr *olsv1alpha1.OLSConfig) (*monv1.PrometheusRule, error) {
+	metaLabels := generateAppServerSelectorLabels()
+	metaLabels["app.kubernetes.io/component"] = "metrics"
+
+	rule := monv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      AppServerPrometheusRuleName,
+			Namespace: r.Options.Namespace,
+			Labels:    metaLabels,
+		},
+		Spec: monv1.PrometheusRuleSpec{
+			Groups: []monv1.RuleGroup{
+				{
+					Name: "ols.operations.rules",
+					Rules: []monv1.Rule{
```

Comment:
Do these rules have a threshold? just trying to understand how this works.

Reply:
what do you mean by threshold?
here are [the operators in PromQL](https://prometheus.io/docs/prometheus/latest/querying/operators/) used in defining recording rules. They are all the calculations available in PromQL.

Reply:
Ohh okay. I guess my question is, whats the difference between having a service monitor to scrape metrics and a recording rule?

Reply:
Understood. Thanks for the clarification.